cpp/models/ode_secir/model.h

```diff
@@ -308,6 +312,247 @@ double get_infections_relative(const Simulation<Base>& sim, double /*t*/, const
     return inf_rel;
 }
 
+/**
+*@brief Computes the reproduction number at a given index time of the Model output obtained by the Simulation.
+*@param t_idx The index time at which the reproduction number is computed.
+*@param sim The Model Simulation.
+*@returns The computed reproduction number at the provided index time.
+*/
+template <class Base>
+IOResult<ScalarType> get_reproduction_number(size_t t_idx, const Simulation<Base>& sim)
```

Comment:
I know/assume that this contains the derivative of the ODE so a lot of code-doubling happens. Generally it is just not a very good a idea to introduce redudant code parts. We should maybe think of how only one responsible code part returns for instance the mechanism of X and then this is used for get_derivatives and get_reproduction_number.